### DISCUSSION TITLE
state not persisted when navigating via next/link

### DISCUSSION THREAD

**Discussion by mrguamos** [vercel/next.js#43968] [Help] [answered]
whenever I navigate using next/link and press the back button of the browser, the input values gets reset.
but when I try to manually type the url in the browser and press back, the values are persisted

how do I persist the state so that whenever I press the back button, the input values are still intact?

_app.tsx

```
import '../styles/globals.css'
import type { AppProps } from 'next/app'

export default function App({ Component, pageProps }: AppProps) {
  return (<Layout><Component {...pageProps} /> </Layout>)
}

type Props = {
  children: React.ReactNode
}

const Layout = ({children} : Props) => {
  return (
    <>
      {children}
    </>
  )
}
```

index.tsx

```
import Head from 'next/head'
import Link from "next/link";

export default function Home() {
  return (
    <>
      <Head>
        <title>Index</title>
        <meta name="description" content="Generated by create-t3-app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <main className="flex min-h-screen flex-col items-center justify-center bg-gradient-to-b from-[#2e026d] to-[#15162c]">
        {/* how to preserve the value of this input field on history back */}
        <input type={'text'} />
        <Link  href={'/about'}>About</Link>
      </main>
    </>
  )
}

```

about.tsx
```
import { type NextPage } from "next";
import Head from "next/head";

const About: NextPage = () => {
  return (
    <>
      <Head>
        <title>About</title>
        <meta name="description" content="Generated by create-t3-app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
    </>
  );
};

export default About;
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

AFAIK, only Chrome, or rather Chromium, have the behaviour your describe.

- https://bugs.chromium.org/p/chromium/issues/detail?id=1057463#makechanges
- https://bugs.chromium.org/p/chromium/issues/detail?id=1258293

I guess, you gotta place the `input` field on the `_app` level if you want to preserve it's state, or save it to session storage. As part of SPA transitions.

It is a bit up in the air whether or not:

> but when I try to manually type the url in the browser and press back, the values are persisted

is not an actual bug... I tested on Safari and FF just now, and they do not preserve it... Brave and Chrome do keep it.

**Reply by mrguamos** [reply]
I see, I found this one https://github.com/AlexSapoznikov/react-next-keep-alive which caches the component just like what Vue's KeepAlive is doing.
